fury/shaders/base.py

```diff
@@ -135,14 +135,41 @@ def add_shader_callback(actor, callback):
     callback : callable
         function or class that contains 3 parameters: caller, event, calldata.
         This callback will be trigger at each `UpdateShaderEvent` event.
+    priority : float
+         Commands with a higher priority are called first.
+
+    Returns:
+    --------
+        id_observer : int
+            An unsigned Int tag which can be used later to remove the event
+            or retrieve the vtkCommand used in the observer.
+            See more at: https://vtk.org/doc/nightly/html/classvtkObject.html
+
+    Example
+
+    ```python
+    add_shader_callback(actor, func_call1)
+    id_observer = add_shader_callback(actor, func_call2)
+    actor.GetMapper().RemoveObserver(id_observer)
+    ```
 
     """
     @vtk.calldata_type(vtk.VTK_OBJECT)
     def cbk(caller, event, calldata=None):
         callback(caller, event, calldata)
 
+    if isinstance(priority, (float, int)) is False:
+        # This avoid the strange error checking from vtk
+        # mapper.AddObserver(vtk.vtkCommand.UpdateShaderEvent, cbk, '12')
+        # TypeError: AddObserver argument 1: string or None required
```

Comment:
you can remove the comment. the `if` above is explicit and the comment is more maintenance in case of a change 😄